### PR TITLE
Use external arrow to compile

### PR DESCRIPTION
**Pull request by zhejiangxiaomai**
Some projects (like gluten) have already compiled arrow before. And we can use external arrow to compile velox.

### CODE REVIEW

File: third_party/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,54 @@ if(VELOX_ENABLE_ARROW)
   else()
     set(THRIFT_SOURCE "BUNDLED")
   endif()
+  # Use external arrow & parquet only if <package>_DIR is defined
```
Comment (rui-mo) [inline third_party/CMakeLists.txt:27]: Let's add `.` at the end for comments.

**Review by rui-mo — COMMENTED** [review]
Also there's some format issue.

File: third_party/CMakeLists.txt (hunk 1 above)
Comment (zhejiangxiaomai) [reply third_party/CMakeLists.txt:27]: Thanks.

**Comment by zhejiangxiaomai** [comment]
> Also there's some format issue.

Yes. Just Fixed it.

File: third_party/CMakeLists.txt [hunk 2]
```diff
@@ -24,6 +24,54 @@ if(VELOX_ENABLE_ARROW)
   else()
     set(THRIFT_SOURCE "BUNDLED")
   endif()
+  # Use external arrow & parquet only if <package>_DIR is defined.
+  if(DEFINED Arrow_HOME)
```
Comment (rui-mo) [inline third_party/CMakeLists.txt:28]: Arrow_HOME -> ARROW_HOME

File: third_party/CMakeLists.txt [hunk 3]
```diff
@@ -24,6 +24,54 @@ if(VELOX_ENABLE_ARROW)
...
+    find_package(Arrow PATHS "${Arrow_HOME}/arrow_install" NO_DEFAULT_PATH)
```
Comment (rui-mo) [inline third_party/CMakeLists.txt:29]: Is arrow_install a path in Arrow, or just a path defined in Gluten?

File: third_party/CMakeLists.txt [hunk 4]
```diff
@@ -24,6 +24,54 @@ if(VELOX_ENABLE_ARROW)
...
+    endif()
+
+    if(Thrift_FOUND)
+      add_library(thrift INTERFACE)
+      target_link_libraries(thrift INTERFACE thrift::thrift)
+      message(STATUS "Using system thrift")
+    else()
+      add_library(thrift STATIC IMPORTED GLOBAL)
+      if(NOT Thrift_FOUND)
+        set(THRIFT_ROOT ${Arrow_HOME}/arrow_ep/cpp/build/thrift_ep-install)
```
Comment (rui-mo) [inline third_party/CMakeLists.txt:57]: Looks arrow_ep and thrift_ep-install are also from Gluten.

**Review by rui-mo — COMMENTED** [review]
Can we desribe

**Comment by rui-mo** [comment]
@zhejiangxiaomai Looks we do not need this PR any more. Gluten will use arrow provided by Velox.

**Pull request closed without merging**